### PR TITLE
Support Want-Digest for external content with Head and Get

### PR DESCRIPTION
**Pull request by lsitu**
Fixes https://jira.duraspace.org/browse/FCREPO-2587

Add support for Head and Get with Want-Digest header to external content.

### CODE REVIEW

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java [hunk 1]
```diff
@@ -44,11 +53,29 @@ private CacheEntryFactory() {
...
+            }
+            return new BinaryCacheEntry(property);
         }
-        return new BinaryCacheEntry(property);
+    }
+
+    /**
+     * Utility method to parse the external body content in format:
+     *   message/external-body; access-type=URL; url="http://www.example.com/file"
+     * @param mimeType
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java]:
```
[WARNING] /fcrepo4/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java:71: warning: no description for @param
[WARNING] * @param mimeType
```

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java [hunk 2]
```diff
@@ -383,15 +384,33 @@ public RdfStream getFixity(final IdentifierConverter<Resource, FedoraResource> i
...
+                final Map<String, String> params = CacheEntryFactory.parseExternalBody(mimeType);
+                final String accessType = params.get("access-type");
+                final String resourceLocation = params.get(accessType.toLowerCase());
+
+                if (accessType != null && accessType.equals("URL")) {
+                    LOGGER.debug("Checking external resource: " + resourceLocation);
+
+                    return CacheEntryFactory.forProperty(getProperty(HAS_MIME_TYPE)).checkFixity(algorithms);
+                } else {
+                    throw new UnsupportedAccessTypeException("Unsupported access-type " + accessType
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java]: I would expect the following command (with JUNK access-type) to fail... but it succeeds:
```
curl -i -ufedoraAdmin:fedoraAdmin -XPUT -H"Content-Type: message/external-body;access-type=JUNK;url=\"https://blognumbers.files.wordpress.com/2010/09/6.jpg\"" localhost:8080/rest/image

```
I see it fails when requesting `Want-Digest`... but should also fail at the point of creation. That can be a new, follow-on ticket.
https://fcrepo.github.io/fcrepo-specification/#external-content

Comment (awoods) [reply fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java]: Nevermind. I see this issue is addressed in: https://jira.duraspace.org/browse/FCREPO-2586

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java [hunk 3]
```diff
@@ -44,11 +53,29 @@ private CacheEntryFactory() {
...
+     * @return Map with key mime-type for MimeType value.
+     */
+    public static Map<String, String> parseExternalBody(final String mimeType) {
+        final Map<String, String> params = new HashMap<String, String>();
+        Splitter.on(';').omitEmptyStrings().trimResults()
+            .withKeyValueSeparator(Splitter.on('=').limit(2)).split("mime-type=" + mimeType.trim())
+            .forEach((k, v) -> params.put(k.toLowerCase(), v.replaceAll("^\"|\"$", "")));
```
[the remark this answers is not in the document]
Comment (lsitu) [reply fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java]: Sure. Comment added in commit https://github.com/fcrepo4/fcrepo4/pull/1238/commits/22c52a57b4d391e0bc1c30b871e0d1a3300d9bcb.

**Comment by lsitu** [comment]
@awoods I've addressed the issues in commit https://github.com/fcrepo4/fcrepo4/pull/1238/commits/22c52a57b4d391e0bc1c30b871e0d1a3300d9bcb. And the PR is rebased.

**Pull request merged**